### PR TITLE
set unique_with in model of domain and project dashboard

### PR DESCRIPTION
**Pull request by stat-kwon**
Signed-off-by: seolmin <seolmin@megazone.com>

### Category
- [ ] New feature
- [ ] Bug fix
- [x] Improvement
- [ ] Refactor
- [ ] etc

### Description
* set unique_with in model of domain and project dashboard

### CODE REVIEW

**Comment by stat-kwon** [comment]
Below is an example of validation that occurs when unique_with is applied.

* Example of DomainDashboard
```python
  File "/usr/local/lib/python3.10/site-packages/spacectl/command/execute.py", line 291, in _call_api
    raise Exception(e.message.strip())
Exception: Tried to save duplicate unique values. (keys = ['name', 'domain_id'])
```

* Example of ProjectDashboard
```python
    for response in response_stream:
  File "/usr/local/lib/python3.10/site-packages/spacectl/command/execute.py", line 291, in _call_api
    raise Exception(e.message.strip())
Exception: Tried to save duplicate unique values. (keys = ['name', 'project_id'])
```

File: src/spaceone/dashboard/model/project_dashboard_model.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def to_dict(self):
 
 class ProjectDashboard(MongoModel):
     project_dashboard_id = StringField(max_length=40, generate_id='project-dash', unique=True)
-    name = StringField(max_length=255)
+    name = StringField(max_length=255, unique_with='project_id')
```
Comment (whdalsrnt) [inline src/spaceone/dashboard/model/project_dashboard_model.py:24]: Need to create compound unique index.
- domain_id
- project_id
- name

Please change code as follows.
`unique_with=['project_id', 'domain_id']`

**Review by whdalsrnt — CHANGES_REQUESTED** [review]
After checking the review, the code needs to be changed.
Thank you.

File: src/spaceone/dashboard/model/domain_dashboard_model.py [hunk 2]
```diff
@@ -59,6 +59,10 @@ class DomainDashboard(MongoModel):
             'viewers',
             'labels',
             'user_id',
-            'domain_id'
+            'domain_id',
+            {
```
Comment (whdalsrnt) [inline src/spaceone/dashboard/model/domain_dashboard_model.py:63]: Setting unique_with as a field property automatically creates a unique index.
So please remove the current index code.

File: src/spaceone/dashboard/model/project_dashboard_model.py [hunk 3]
```diff
@@ -65,6 +65,10 @@ class ProjectDashboard(MongoModel):
             'labels',
             'project_id',
             'user_id',
-            'domain_id'
+            'domain_id',
+            {
```
Comment (whdalsrnt) [inline src/spaceone/dashboard/model/project_dashboard_model.py:69]: Setting unique_with as a field property automatically creates a unique index.
So please remove the current index code.

**Review by whdalsrnt — CHANGES_REQUESTED** [review]

**Pull request merged**